jdbc/src/main/java/io/kestra/jdbc/runner/JdbcExecutor.java

```diff
@@ -227,13 +231,21 @@ public JdbcExecutor(
         final FlowRepositoryInterface flowRepository,
         final AbstractJdbcFlowTopologyRepository flowTopologyRepository,
         final ApplicationEventPublisher<ServiceStateChangeEvent> eventPublisher,
-        final TracerFactory tracerFactory
+        final TracerFactory tracerFactory,
+        final ExecutorsUtils executorsUtils
         ) {
         this.serviceLivenessCoordinator = serviceLivenessCoordinator;
         this.flowRepository = flowRepository;
         this.flowTopologyRepository = flowTopologyRepository;
         this.eventPublisher = eventPublisher;
         this.tracer = tracerFactory.getTracer(JdbcExecutor.class, "EXECUTOR");
+
+        // By default, we start half-available processors count threads with a minimum of 4 by executor service
+        // for the worker task result queue and the execution queue.
+        // Other queues would not benefit from more consumers.
+        this.numberOfThreads = threadCount != 0 ? threadCount : Math.max(4, Runtime.getRuntime().availableProcessors() / 2);
```

Comment:
This should not be `Math.min(4, Runtime.getRuntime().availableProcessors() / 2);`  or the comment is wrong?